### PR TITLE
Create partition

### PR DESCRIPTION
**Pull request by Tumao727**
complete #9 

create partition

### CODE REVIEW

File: client/src/hooks/Pagination.ts [hunk 1]
```diff
@@ -1,14 +1,26 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 
 const PAGE_SIZE = 15;
-export const usePaginationHook = () => {
+export const usePaginationHook = (list: any[]) => {
   const [offset, setOffset] = useState(0);
   const [currentPage, setCurrentPage] = useState(0);
   const [total, setTotal] = useState(0);
+  const [data, setData] = useState<any[]>([]);
```
Comment (nameczz) [inline client/src/hooks/Pagination.ts:8]: usememo instead of usestate to control data and offset

**Review by shanghaikid — APPROVED** [review]

File: client/src/pages/partitions/partitions.tsx [hunk 2]
```diff
@@ -94,12 +130,27 @@ const Partitions: FC<{ data: PartitionView[]; loading: boolean }> = ({
...
+  const handleCreatePartition = async (name: string) => {
+    const param: PartitionManageParam = {
+      partitionName: name,
+      collectionName,
+      type: 'create',
+    };
+
+    await PartitionHttp.createPartition(param);
+
+    openSnackBar('create partition success');
```
Comment (nameczz) [inline client/src/pages/partitions/partitions.tsx:142]: i18n

File: client/src/types/Common.ts [hunk 3]
```diff
@@ -20,3 +20,5 @@ export interface IPaginationRes {
   offset: number;
   total_count: number;
 }
+
+export type ManageRequestMethods = 'delete' | 'create';
```
Comment (nameczz) [inline client/src/types/Common.ts:24]: use enum?

File: client/src/hooks/Pagination.ts [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
   const [currentPage, setCurrentPage] = useState(0);
-  const [total, setTotal] = useState(0);
+
+  const offset = useMemo(() => PAGE_SIZE * currentPage, [currentPage]);
+  const total = list.length;
+  const data = useMemo(() => {
+    const offset = PAGE_SIZE * currentPage;
+    const test = list.slice(offset, offset + PAGE_SIZE);
+    return test;
+  }, [list, currentPage]);
```
Comment (nameczz) [inline client/src/hooks/Pagination.ts:13]: put offset total data in one useMemo and rename test

**Review by nameczz — APPROVED** [review]

**Pull request merged**